docs/content/common-subs/check-kubestellar-helm-deployment-running.md

```diff
@@ -0,0 +1,11 @@
+<!--check-kubestellar-helm-deployment-running-start-->
+```shell
+echo -n 'Waiting for KubeStellar to be ready'
+while ! kubectl exec $(kubectl get pod --selector=app=kubestellar -o jsonpath='{.items[0].metadata.name}' -n kubestellar) -n kubestellar -c init -- ls /home/kubestellar/ready &> /dev/null; do
```

Comment:
The two `kubectl` commands here are sensitive to the setting of `$KUBECONFIG`. Since this document generally uses an explicit setting of that envar on each command line, that could be done here. One way would be very direct:

```
while ! KUBECONFIG=~/.kube/config kubectl exec $(KUBECONFIG=~/.kube/config kubectl get pod --selector=app=kubestellar -o jsonpath='{.items[0].metadata.name}' -n kubestellar) -n kubestellar -c init -- ls /home/kubestellar/ready &> /dev/null; do
```

Another approach would be to break out the discovery of the pod name into a command that is done once at the start; it does not have to be rediscovered every time around the loop. But we _do_ have to wait for the pod to exist, which is covered by waiting for the pod to be ready.

```
KUBECONFIG=~/.kube/config kubectl wait -n kubestellar -for condition=ready pod --selector app=kubestellar
KS_CORE_POD=$(KUBECONFIG=~/.kube/config kubectl get pod --selector=app=kubestellar -o jsonpath='{.items[0].metadata.name}' -n kubestellar)
while ! KUBECONFIG=~/.kube/config kubectl exec $KS_CORE_POD -n kubestellar -c init -- ls /home/kubestellar/ready &> /dev/null; do
...
done
```

Reply:
Actually, I that`kubectl wait` for the pod to be ready might NOT do what we want if invoked before the pod exists. I had that fail immediately with a complaint about "no resource found" when that case happened to me recently.